internal/engine/upper.go

```diff
@@ -405,6 +422,39 @@ func handleGlobalYAMLApplyError(
 	state.GlobalYAMLState.LastError = event.Error
 }
 
+func handleTiltfileReloaded(
+	ctx context.Context,
+	state *store.EngineState,
+	event TiltfileReloadedAction,
+) {
+	manifests := event.Manifests
+	globalYAML := event.GlobalYAML
+	err := event.Err
+	if err != nil {
+		logger.Get(ctx).Infof("Unable to parse Tiltfile: %v", err)
+
+		for _, ms := range state.ManifestStates {
+			ms.LastManifestLoadError = err
+		}
+		return
+	}
+	newDefOrder := make([]model.ManifestName, len(manifests))
+	for i, m := range manifests {
+		state.ManifestStates[m.ManifestName()].LastManifestLoadError = nil
+		oldManifest := state.ManifestStates[m.ManifestName()].Manifest
+
+		newDefOrder[i] = m.ManifestName()
+		if !oldManifest.Equal(m) {
+			newManifestState := store.NewManifestState(m)
+			state.ManifestStates[m.ManifestName()] = newManifestState
```

Comment:
Replacing the whole `ManifestState` makes me a bit nervous - is there stuff there we explicitly want to throw away, or could we get away with only replacing the existing state's `Manifest` field?

e.g., IIUC, if this event comes in while the manifest is queued with pending changes, we're now discarding that state (along with last build result and pod info etc)

Reply:
Good question, I can play around with it! I'm not sure.

Reply:
Yeah good call this works!